### PR TITLE
[WIFI-3232] Fix Docker Compose deployment

### PR DESCRIPTION
**Pull request by oblom0v**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,21 +38,25 @@ RUN cmake --build . --config Release -j8
...
 
-EXPOSE 15002
-EXPOSE 16002
-EXPOSE 16003
-EXPOSE 17002
-EXPOSE 16102
+EXPOSE 15002 16002 16003 17002 16102
+
+COPY docker-entrypoint.sh /
+RUN chmod +x /docker-entrypoint.sh
```
Comment (dunaev-opsfleet) [inline Dockerfile:59]: You can get rid of this line if you make `docker-entrypoint.sh` executable in the repository itself as permissions are also copied with file

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+set -e
+
+if [ "$1" = '/ucentral/ucentralgw' -a "$(id -u)" = '0' ]; then
+    chown -R "$UCENTRALGW_USER". "$UCENTRALGW_ROOT" "$UCENTRALGW_CONFIG"
```
Comment (dunaev-opsfleet) [inline docker-entrypoint.sh:5]: What for is the dot after UCENTRALGW_USER?

File: Dockerfile (hunk 1 above)
Comment (oblom0v) [reply Dockerfile:59]: You're right, thought I set it to executable and Git didn't keep the permissions, but this was on another machine.

File: docker-entrypoint.sh (hunk 2 above)
Comment (oblom0v) [reply docker-entrypoint.sh:5]: This is to apply the same group ownership (basically just leaving the group blank). But it seems that the dot is an old way of separating the user and group fields and chown just supports it out of legacy reasons, so I think I'll change it to a colon.

**Comment by oblom0v** [comment]
Ignore the failing docker-compose CI check for now, it will be successful when https://github.com/Telecominfraproject/wlan-cloud-ucentralsec/pull/7 gets merged.

**Comment by brennerm** [comment]
Do you know the reason why the build is failing?

**Comment by oblom0v** [comment]
@brennerm Yes the current `main` image of ucentralsec executes as a non-priviliged user, and the Compose deployment uses a bind mount, so the application inside the container has no write permissions in the folder and therefore can not write the SQLite database. This wasn't the case in the previous commit since at this time the switch to SQLite in the deploy repository wasn't merged yet.

**Review by dunaev-opsfleet — APPROVED** [review]

**Pull request merged**